lib/bolt/module_installer/puppetfile.rb

```diff
@@ -97,20 +97,43 @@ def assert_satisfies(specs)
           end
         end
 
-        return if unsatisfied_specs.empty?
-
+        #versionless_mods = @modules.any? do |mod|    
+        #  mod.version.nil? 
+        #end
+        versionless_mods = []
+        @modules.each {|mod|
+          if mod.version.nil?
+            versionless_mods << mod
+          end
+        }
+        
+        return if unsatisfied_specs.empty? && versionless_mods.empty?
         command = Bolt::Util.windows? ? 'Install-BoltModule -Force' : 'bolt module install --force'
+        
+        if !unsatisfied_specs.empty?
```

Comment:
```suggestion
        if unsatisfied_specs.any?
```

Prefer not to negate in a conditional, since it can make the logic a bit harder to understand. Another alternative is `unless unsatisfied_spec.empty?`.